src/uris/deepLinks/deepLinkService.ts

```diff
@@ -1268,6 +1312,16 @@ export class DeepLinkService implements Disposable {
 							break;
 						}
 
+						// Storing link info in case the switch causes a new window to open
+						await this.container.storage.store('deepLinks:pending', {
```

Comment:
Note: this part is definitely iffy. One issue with storing it like this before knowing if the user will open a new window or not, is that the user can 1. open a new window to complete the switch process, or 2. cancel the switch process altogether. In the case of 2, we need to clear out the stored deep link, but the service currently doesn't distinguish between the two, so the pending link doesn't get cleared out properly. In that case, the next time you open a window, it tries to continue the link by opening GitLens inspect, and gives an error if the repo wasn't found.

If we have a way of distinguishing the result of the "switch" here, we can instead use that result to determine if we need to store the pending link at all, and guarantee that we don't store it if the user did not choose to complete the switch in a new window.

Reply:
Looks like this is the best way for now, as eventing from the quickpick step is way trickier and hackier

Reply:
@axosoft-ramint take a look at:

https://github.com/gitkraken/vscode-gitlens/blob/3e248d338e1025e0054e7d6bcdebf5302b3b3fb0/src/git/actions/worktree.ts#L11-L39

I added this crude "result" completion that could be passed into the `worktree` command -- we could do something similar for `switch`. It could also be the inverse -- pass in a callback to get run or something like that.